### PR TITLE
PRSQC WDL First Iteration

### PR DESCRIPTION
**Pull request by kockan**
- Satisfies basic requirements (`prs_score` and `full_model_score` thresholds/boundaries)
- Allows manual passing of controls
- Nice visual PCA QC output(s) which can be easily displayed on the google bucket page
- Input file format(s)/header(s)/etc. subject to change in future iterations

### CODE REVIEW

File: Utilities/WDLs/PRSQC.wdl [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+        # Load alphashape
+        with open("~{alphashape}", "rb") as infile:
+            alpha_shape = pickle.load(infile)
+
+        # Set the distance threshold
+        dist_thresh = ~{distance_threshold}
+
+        # Read test data
+        # Expected input format: tsv with columns PRS_SCORE, PC1, PC2, and FULL_MODEL_SCORE
+        # There is no header at the moment but it can be added to the format
```
Comment (kachulis) [inline Utilities/WDLs/PRSQC.wdl:179]: yes, let's assume there will be a header and read column names from that

File: Utilities/WDLs/PRSQC.wdl [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+    call DetectPCANovelties as DetectPCANoveltiesSample {
+        input:
+            test_sample = prs_sample,
+            alphashape = alphashape,
+            cpu = cpu,
+            mem_gb = mem_gb,
+            disk_size_gb = disk_size_gb
+    }
+
+    call FinalizeQCOutputs {
```
Comment (kachulis) [inline Utilities/WDLs/PRSQC.wdl:60]: Do we need to spin up a whole new vm for this?  can't this just be 
```
qc_passed = qc_passed_control && qc_passed_sample && pca_qc_passed_control && pca_qc_passed_sample
```

**Review by kachulis — CHANGES_REQUESTED** [review]
Looks good, thanks @kockan!  A few restructuring comments.

File: Utilities/WDLs/PRSQC.wdl [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+            input:
+                scores = prs_control,
+                thresholds = control_thresholds,
+                cpu = cpu,
+                mem_gb = mem_gb,
+                disk_size_gb = disk_size_gb,
+                docker = docker
+        }
+
+        call DetectPCANovelties as DetectPCANoveltiesControl {
```
[the remark this answers is not in the document]
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: Done

File: Utilities/WDLs/PRSQC.wdl [hunk 4]
```diff
@@ -0,0 +1,268 @@
...
+        Boolean manually_passed_control = false
+    }
+
+    Int cpu = 1
+    Int mem_gb = 4
+    Int disk_size_gb = 32
+    String docker = "us.gcr.io/broad-dsde-methods/python-data-slim:1.0"
+
+    if(!manually_passed_control) {
+        call CheckThresholds as CheckControl {
```
[the remark this answers is not in the document]
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: Done

File: Utilities/WDLs/PRSQC.wdl [hunk 5]
```diff
@@ -0,0 +1,268 @@
...
+
+    String output_basename = basename(scores, ".tsv")
+
+    command <<<
+        set -euo pipefail
+
+        cat <<'EOF' > script.py
+        import pandas as pd
+
+        scores = pd.read_csv('~{scores}', sep = '\t', header = None, names = ["prs_score", "pc1", "pc2", "full_model_score"])
```
[the remark this answers is not in the document]
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: Done

File: Utilities/WDLs/PRSQC.wdl [hunk 6]
```diff
@@ -0,0 +1,268 @@
...
+        thresholds = pd.read_csv('~{thresholds}', sep = '\t', header = None, names = ["min", "max"])
```
[the remark this answers is not in the document]
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: Done

File: Utilities/WDLs/PRSQC.wdl [hunk 7]
```diff
@@ -0,0 +1,268 @@
...
+
+        # Confirm scores are within the boundaries
+        prs_score_passed = True
+        if scores.iloc[0,0] < thresholds.iloc[0,0] or scores.iloc[0,0] > thresholds.iloc[0,1]:
```
[the remark this answers is not in the document]
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: Done

File: Utilities/WDLs/PRSQC.wdl (hunk 1 above)
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: Done

File: Utilities/WDLs/PRSQC.wdl (hunk 2 above)
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: We certainly can. I might have overcomplicated this. Done

File: Utilities/WDLs/PRSQC.wdl [hunk 8]
```diff
@@ -0,0 +1,268 @@
+version 1.0
+
+# Simple PRS QC wdl for the PROGRESS VA project
+workflow PRSQC {
+    input {
+        String sample_name
+        File prs_control
+        File prs_sample
+        File control_thresholds
+        File sample_thresholds
```
[the remark this answers is not in the document]
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: Done

File: Utilities/WDLs/PRSQC.wdl [hunk 9]
```diff
@@ -0,0 +1,268 @@
...
+        File alphashape
```
[the remark this answers is not in the document]
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl:20]: Done

**Comment by kockan** [comment]
Ready for round-2

File: Utilities/WDLs/PRSQC.wdl [hunk 10]
```diff
@@ -18,30 +27,32 @@ workflow PRSQC {
...
                 cpu = cpu,
                 mem_gb = mem_gb,
-                disk_size_gb = disk_size_gb
+                disk_size_gb = disk_size_gb,
+                docker = docker
         }
     }
 
-    call CheckThresholds as CheckSample {
+    call CheckScoresAgainstExpectedValues as CheckScoresAgainstExpectedSampleValues {
```
[the remark this answers is not in the document]
Comment (kockan) [reply Utilities/WDLs/PRSQC.wdl]: Done

**Review by kachulis — APPROVED** [review]
Looks good, thanks @kockan!

**Pull request merged**